lib/scss_lint/config.rb

```diff
@@ -178,6 +178,16 @@ def initialize(options)
       validate_linters
     end
 
+    def load_plugins
+      load_plugins_and_merge_config.tap { ensure_plugins_have_default_options }
+    end
+
+    def load_plugins_and_merge_config
+      Plugins.new(@options).load.each do |plugin|
+        @options = self.class.send(:smart_merge, plugin.config_options, @options)
```

Comment:
The order that we're loading configurations here requires some closer attention. With this implementation, if I want to share a gem that sets configuration for linters included in the main `scss-lint` project (e.g. sets standard `Indentation` width or a maximum `NestingDepth`) that can be used across all projects in my organization, I can't.

This is because the default configuration that is automatically loaded with every configuration will override the plugin configuration here.

This suggests that this plugin loading code needs to be shimmed somewhere in the actual configuration loading process _before_ the default configuration is merged. This preserves the load order of `default` → `plugins` → `local .scss-lint.yml`. I've implemented this in a382abf.